City-of-Helsinki/youth-membership-ui
# YM-402 | Fix mobile navigation by adopting navigation component from HDS

## Description

This change replaces our own navigation implementation with that of HDS. By doing this, we are able to tackle a few accessibility issues.
* The menu mobile menu becomes usable with screen readers
* A skip to content link will be added
* The problematic labelling of the logo and service name will be fixed

## Context



[YM-402](https://helsinkisolutionoffice.atlassian.net/browse/YM-402)

## How Has This Been Tested?

I've ran the test suite against these changes and checked that it passes. I've manually tested the new menu.

## Manual Testing Instructions for Reviewers

With the mobile layout and screen reader

1. Expect the mobile menu to have an accessible name
2. Expect the mobile menu to have aria roles which denote whether it is open or not
3. Expect the mobile menu content to be accessible

With a screen reader

1. Expect the logo image and application title to have a single shared label ("Helsinki: Jässäri")

With a screen reader

1. Expect the first focusable element to be a jump to content button
2. Expect the button to skip the menu and jump to content

With a screen reader

1. Expect language items to have correct lang attributes

With a browser

1. Start creating a new profile
2. Expect to see a sign out button
3. Fill in details
4. Open approval mail
5. Expect user menu to be hidden
6. Expect logo and app title to lead to the same page

With a browser

1. Sign in
2. Expect user menu to contain link to Helsinki-profile and logging out
